File: appMain/src/main/java/com/cartlc/tracker/server/DCPost.java
package com.cartlc.tracker.server;

import android.util.Log;

import com.cartlc.tracker.app.TBApplication;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;

import de.greenrobot.event.EventBus;

/**
 * Created by dug on 8/24/17.
 */

public class DCPost {

    static final String TAG = "DCPost";

    protected String getResult(HttpURLConnection connection) throws IOException {
        InputStream inputStream;
        try {
            inputStream = connection.getInputStream();
        } catch (Exception ex) {
            showError(connection);
            return null;
        }
        if (inputStream == null) {
            showError(connection);
            return null;
        }
        return getStreamString(inputStream);
    }

    protected String getStreamString(InputStream inputStream) throws IOException {
        if (inputStream == null) {
            return null;
        }
        BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
        StringBuilder sbuf = new StringBuilder();
        String inputLine;
        while ((inputLine = reader.readLine()) != null) {
            if (sbuf.length() > 0) {
                sbuf.append("\n");
            }
            sbuf.append(inputLine);
        }
        reader.close();
        return sbuf.toString();
    }

    protected void showError(HttpURLConnection connection) throws IOException {
        final String errorMsg = getStreamString(connection.getErrorStream());
        String msg;
        if (errorMsg == null) {
            msg = "Server might be DOWN. Try again later.";
        } else {
            msg = "Server COMPLAINT: " + errorMsg;
        }
        Log.e(TAG, msg);
        TBApplication.ShowError(msg);
    }
}

File: appMain/src/main/java/com/cartlc/tracker/model/server/DCPost.kt
/**
 * Copyright 2018, FleetTLC. All rights reserved
 */
package com.cartlc.tracker.model.server

import android.util.Log

import com.cartlc.tracker.ui.app.TBApplication

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.io.InputStreamReader
import java.net.HttpURLConnection

/**
 * Created by dug on 8/24/17.
 */
open class DCPost {


    companion object {

        private const val TAG = "DCPost"
        private const val MAX_SIZE = 65536

    }


    @Throws(IOException::class)
    protected fun getResult(connection: HttpURLConnection): String? {
        val inputStream: InputStream?
        try {
            inputStream = connection.inputStream
        } catch (ex: Exception) {
            showError(connection, ex.message)
            return null
        }
        if (inputStream == null) {
            showError(connection, null)
            return null
        }
        return getStreamString(inputStream)
    }

    @Throws(IOException::class)
    protected fun getStreamString(inputStream: InputStream?): String? {
        if (inputStream == null) {
            return null
        }
        val buffer = CharArray(1024)
        val reader = BufferedReader(InputStreamReader(inputStream)!!)
        var count: Int
        val sbuf = StringBuffer()
        while (true) {
            count = reader.read(buffer, 0, buffer.size)
            if (count < 0) {
                break
            }
            sbuf.append(buffer.copyOfRange(0, count))
            if (sbuf.length > MAX_SIZE) {
                break
            }
        }
        reader.close()
        return sbuf.toString()
    }

    @Throws(IOException::class)
    protected fun showError(connection: HttpURLConnection, message: String?) {
        val errorMsg = getStreamString(connection.errorStream)
        val msg: String
        if (errorMsg == null) {
            if (message == null) {
                msg = "Server might be DOWN. Try again later."
            } else {
                msg = "Server connection error: $message"
            }
        } else {
            msg = "Server COMPLAINT: $errorMsg"
        }
        Log.e(TAG, msg)
        TBApplication.ShowError(msg)
    }

}
